TestPostLede fails

Trying to package `writefreely` for Alpine Linux (busybox/musl libc)
Build is OK but Tests fail at :

```
=== RUN   TestPostLede
    TestPostLede: posts_test.go:34: #2: Got 'Sup? Everyone says punctuation is punctuation.' from 'Sup? Everyone says punctuation is punctuation.'; expected 'Sup?'
--- FAIL: TestPostLede (0.00s)
```

Packaging is using :
 
```
make build
make test
```

Requirements are `go, go-bindata, sqlite, nodejs-less, nodejs-less-plugin-clean-css"

**Version or last commit**:
  - `v0.12.0`

--- a/oauth_test.go
+++ b/oauth_test.go
@@ -244,7 +244,7 @@ func TestViewOauthCallback(t *testing.T) {
 		req, err := http.NewRequest("GET", "/oauth/callback", nil)
 		assert.NoError(t, err)
 		rr := httptest.NewRecorder()
-		err = h.viewOauthCallback(nil, rr, req)
+		err = h.viewOauthCallback(&App{cfg: app.Config(), sessionStore: app.SessionStore()}, rr, req)
 		assert.NoError(t, err)
 		assert.Equal(t, http.StatusTemporaryRedirect, rr.Code)
 	})

--- a/parse/posts.go
+++ b/parse/posts.go
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2018 A Bunch Tell LLC.
+ * Copyright © 2018-2020 A Bunch Tell LLC.
  *
  * This file is part of WriteFreely.
  *
@@ -53,6 +53,11 @@ func PostLede(t string, includePunc bool) string {
 		t = t[:punc+iAdj]
 	}
 	punc = stringmanip.IndexRune(t, '。')
+	if punc > -1 {
+		c := []rune(t)
+		t = string(c[:punc+iAdj])
+	}
+	punc = stringmanip.IndexRune(t, '?')
 	if punc > -1 {
 		c := []rune(t)
 		t = string(c[:punc+iAdj])

